src/Adapters/BlobAdapter.ts

```diff
@@ -36,16 +40,54 @@ export default class BlobAdapter implements IBlobAdapter {
     protected containerResourceId: string; // resource scope
     protected blobAuthService: IAuthService;
     protected blobProxyServerPath: string;
+    protected useBlobProxy: boolean;
 
     constructor(
         blobContainerUrl: string,
         authService: IAuthService,
-        blobProxyServerPath = '/proxy/blob'
+        blobProxyServerPath = '/proxy/blob',
+        useBlobProxy = true
     ) {
         this.setBlobContainerPath(blobContainerUrl);
         this.blobAuthService = authService;
         this.blobAuthService.login();
         this.blobProxyServerPath = blobProxyServerPath;
+        /**
+         * Check if class has been initialized with CORS enabled or if origin matches dev or prod explorer urls,
+         * override if CORS is forced by feature flag
+         *  */
+        this.useBlobProxy =
+            (useBlobProxy || !validateExplorerOrigin(window.origin)) &&
+            !(
+                localStorage.getItem(
+                    LOCAL_STORAGE_KEYS.FeatureFlags.Proxy.forceCORS
+                ) === 'true'
+            );
+    }
+
+    getBlobContainerURL() {
+        return this.storageAccountHostName && this.containerName
+            ? `https://${this.storageAccountHostName}/${this.containerName}`
+            : '';
+    }
+
+    generateBlobUrl(path: string) {
+        if (this.useBlobProxy) {
+            return `${this.blobProxyServerPath}${path}`;
+        } else {
+            return `${this.getBlobContainerURL()}${path}`;
+        }
+    }
+
+    generateBlobHeaders(headers: AxiosObjParam = {}) {
```

Comment:
more dupe code makes me saaaaaaad. We should make this a base class.

Reply:
BlobHeaders != ADTHeaders even if we made this a base class this method would be "duped"